### DISCUSSION TITLE
Prefect Integration with Ansible

### DISCUSSION THREAD

**Discussion by gaus1986** [PrefectHQ/prefect#3290] [Q&A]
Hi All,

I there a way to integrate Prefect with Ansible. eg use prefect to describe  the workflow/process, then use Ansible to perform portions of the tasks (eg changes to switches, etc).

If there is then, I would be happy to here thoughts on how to.
Thanks

**Comment by mianos** [comment]
You can use AWX to provide a REST interface to playbooks. You also get the authorisation injection and logging as a bonus. It is an idea I have tried where I am. You can also use the ansible API to get an inventory and run playbooks using the API, but you don't get authorisation injection. At scale I am going to just run the lower level scripts with prefect as AWX is a 'happy path' stack and does not deal with failures as well as Prefect.
